packages/pegasus/src/pegasus.js

```diff
@@ -273,287 +54,343 @@ const makePegasus = (zcf, board, namesByAddress) => {
   };
 
   /**
-   * @type {Store<Peg, Connection>}
+   * @type {LegacyWeakMap<Peg, LocalDenomState>}
    */
-  const pegToConnection = makeStore('Peg');
+  const pegToDenomState = makeLegacyWeakMap('Peg');
 
   /**
    * Create a fresh Peg associated with a descriptor.
    *
-   * @typedef {Object} PegDescriptor
+   * @typedef {Object} PegasusDescriptor
    * @property {Brand} localBrand
-   * @property {DenomUri} denomUri
+   * @property {Denom} remoteDenom
    * @property {string} allegedName
    *
-   * @param {Connection} c
-   * @param {PegDescriptor} desc
-   * @param {Set<Peg>} pegs
+   * @param {LocalDenomState} state
+   * @param {PegasusDescriptor} desc
    * @returns {Peg}
    */
-  const makePeg = (c, desc, pegs) => {
+  const makePeg = (state, desc) => {
     /** @type {Peg} */
-    const peg = Far('peg', {
+    const peg = Far(`${desc.allegedName} peg`, {
       getAllegedName() {
         return desc.allegedName;
       },
       getLocalBrand() {
         return desc.localBrand;
       },
-      getDenomUri() {
-        return desc.denomUri;
+      getRemoteDenom() {
+        return desc.remoteDenom;
       },
     });
 
-    pegs.add(peg);
-    pegToConnection.init(peg, c);
-    updater.updateState([...pegToConnection.keys()]);
+    pegToDenomState.init(peg, state);
     return peg;
   };
 
+  /**
+   * @param {Object} param0
+   * @param {ReturnType<typeof makeCourierMaker>} param0.makeCourier
+   * @param {LocalDenomState} param0.localDenomState
+   * @param {ERef<TransferProtocol>} param0.transferProtocol
+   * @returns {PegasusConnectionActions}
+   */
+  const makePegasusConnectionActions = ({
+    makeCourier,
+    localDenomState,
+    transferProtocol,
+  }) => {
+    let checkAbort = () => {};
+
+    /** @type {Set<Peg>} */
+    const pegs = new Set();
+
+    /** @type {PegasusConnectionActions} */
+    const pegasusConnectionActions = {
```

Comment:
I prefer to do the `Far` in place, where the object literal appears, unless there's a reason not to. That way, I know there is no unnecessarily mutable object floating around.

I checked, and this change passes both `yarn lint` and `yarn test`, so should be good.
```suggestion
    const pegasusConnectionActions = Far('pegasusConnectionActions', {
```
This becomes more important with the coming `Far` wrappers, as the far object has protected methods wrapping (what will be) the unprotected methods of the object literal.